tests/foreman/ui/test_search.py

```diff
@@ -0,0 +1,175 @@
+    """User with Auditor role. This user has significantly limited menu access."""
+    auditor_role = module_target_sat.api.Role().search(query={'search': 'name="Auditor"'})[0]
+    password = gen_string('alphanumeric')
+    user = module_target_sat.api.User(
+        admin=False,
+        location=[module_location],
+        organization=[module_org],
+        role=[auditor_role],
+        password=password,
+    ).create()
+    user.password = password
+
+    yield user
+
+    user.delete()
+
+
+@pytest.mark.tier2
+def test_positive_vertical_navigation_search_end_to_end(
+    module_target_sat,
+    module_org,
+    module_location,
+    default_org,
+    default_location,
+    auditor_module_user,
+    test_name,
+):
+    """Test the search function of the vertical navigation menu.
+
+    :id: 87660a22-996b-11ee-b8a5-000c2989e153
+
+    :Setup:
+        Create a non-admin user with Auditor role.
+        Create a custom organization and location.
+
+    :Steps:
+        1. Perform a search in the vertical navigation menu with following inputs:
+            1a. valid characters
+            1b. invalid characters
+            1c. current user organization and location
+            1d. custom organization and location not assigned to the user
+        2. Perform the actions as:
+            - admin user
+            - non-admin user with Auditor role
+
+    :expectedresults:
+        Admin user:
+            1a. should find all matching results
+            1b. should not find any results
+            1c. should find both results
+            1d. should find both results
+        User with Auditor role:
+            1a. should find all matching results
+            1b. should not find any results
+            1c. should find both results
+            1d. should not find any results
+
+    :CaseLevel: Integration
+    """
+    valid_characters = list(' ')
+    invalid_characters = list('.,+=/?!@#$%^&*()[]{}|\\\'"')
+    admin_menu_items = [
+        '-',
+        'dashboard',
+        'facts',
+        'subscriptions',
+        'product',
+        'content',
+        'sync plan',
+        'host',
+        'role',
+        'compute',
+        'domain',
+        'user',
+        'settings',
+    ]
+    auditor_menu_items = [
+        'audits',
+        'job',
+        'tasks',
+        'bookmark',
+        'about',
+    ]
+
+    with module_target_sat.ui_session() as session:
+        search = session.host.search_menu
+
+        results = search(' ')
+        assert len(results) == 10, f'Maximum number of search results is 10, got {len(results)}!'
+
+        valid_search_strings = [
+            *valid_characters,
+            *admin_menu_items,
+            *auditor_menu_items,
+            default_org.name,
+            default_location.name,
+            module_org.name,
+            module_location.name,
+        ]
+        menu_search_should_find(search, valid_search_strings)
+
+        menu_search_should_not_find(search, invalid_characters)
```

Comment:
Why only invalid characters? Search for some queries consisting of valid characters that you however don't expect to be found.